src/python/espressomd/interactions.py

```diff
@@ -350,6 +352,81 @@ def default_params(self):
 
         """
         return {}
+    
+    def set_params(self, **kwargs):
+        """Set new parameters.
+
+        """
+        params = self.default_params()
+        params.update(kwargs)
+
+        if "energy" not in params or "force" not in params:
+            required_keys = {"min", "max", "steps", "f"}
+            missing_keys = required_keys - params.keys()
+            if missing_keys:
+                raise ValueError(
+                    f"Missing keys for table generation: {missing_keys}. "
+                    f"To generate energy and force tables, provide {required_keys}."
+                )
```

Comment:
I'm a bit uneasy about dynamically changing the set of expected arguments of a function depending on whether an argument is missing. From a user experience perspective, it's unclear what the function actually wants, or what happens when forces, energies, and a function are provided simultaneously. This also introduces a maintenance issue: the logic from the parent class `set_params()` method is duplicated here, thus if the parent class behavior changes, it won't be reflected here and `TabulatedNonBonded` will diverge from the rest of the code.

What would you think about creating a static class method like `make_tables(cls, f, min, max, steps)` that would return the energy and force tables, which can then be fed to `set_params(self, **kwargs)`?